### PR TITLE
App install behind a Proxy

### PR DESCRIPTION
**Pull request by instance-pro**
https://github.com/owncloud/core/issues/14066

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Thanks a lot @hlop - I'm questioning if we can cover this with unit tests - hmmmm ..

:+1:

**Comment by LukasReschke** [comment]
@owncloud-bot This is okay to test

File: lib/private/installer.php [hunk 1]
```diff
@@ -241,7 +241,7 @@ public static function downloadApp($data = array()) {
 			if(!isset($data['href'])) {
 				throw new \Exception($l->t("No href specified when installing app from http"));
 			}
-			copy($data['href'], $path);
+			file_put_contents($path, \OC_Util::getUrlContent($data['href']));
```
Comment (LukasReschke) [inline lib/private/installer.php]: Please use `\OC::$server->getHTTPHelper()->getUrlContent($data['href'])` instead as `OC_Util::getUrlContent` is deprecated.

**Comment by LukasReschke** [comment]
Let's get this in :+1:

**Comment by DeepDiver1975** [comment]
@karlitschek backport to for 8.0.1 might make sense on this - also fixes https://github.com/owncloud/core/issues/14192 - THX

**Comment by DeepDiver1975** [comment]
preparing pr - seems like this really fixed the issue - https://github.com/owncloud/core/issues/14192#issuecomment-74245436

**Comment by karlitschek** [comment]
yes. please backport

**Pull request merged**